Patches/PlayerControlPatch.cs

```diff
@@ -1310,6 +1316,40 @@ public static void Postfix(PlayerControl __instance, [HarmonyArgument(0)] Player
             PlagueDoctor.OnAnyMurder();
         }
 
+
+
+        // ========================= GHOST ASSIGN PATCH ========================
+
+        var GetKillerRole = killer.GetCustomRole();
+        bool IsImpostor = CustomRolesHelper.IsImpostor(GetKillerRole);
+        bool IsNeutral = CustomRolesHelper.IsNeutral(GetKillerRole);
+        bool IsGhostRole = CustomRolesHelper.IsGhostRole(GetKillerRole);
+
+        if (!IsImpostor && !IsImpostor && !IsGhostRole)
+        {
+            System.Random random = new System.Random(); // Don't worry 😂 I will work on a better system, this is temporary
+            //int getrand = random.Next(0);
+            int num = 0;
+
+            switch (num)
+            {
+                case 0: // (Ghost)Warden
+                    killer.SetRole(RoleTypes.GuardianAngel);
+                    killer.RpcSetCustomRole(CustomRoles.Warden);
+                    break;
+
+                default:
+                    break;
+
+            }
+
+        }
+
```

Comment:
If this will be causing issues, I will make a custom harmony checkdeath prefix patch